recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,202 @@
+            "FLANN_ROOT": self.deps_cpp_info["flann"].rootpath,
+            "BOOST_ROOT": self.deps_cpp_info["boost"].rootpath,
+            "BUILD_tools": self.options.with_tools,
+            "WITH_LIBUSB": self.options.with_usb,
+            "WITH_PNG": self.options.with_png,
+            "WITH_QHULL": self.options.with_qhull,
+            "WITH_CUDA": self.options.with_cuda,
+            "WITH_VTK": self.options.with_vtk,
+            "WITH_PCAP": self.options.with_pcap,
+            "WITH_OPENGL": self.options.with_opengl,
+            "WITH_OPENNI": self.options.with_openni,
+            "WITH_OPENNI2": self.options.with_openni2,
+            "WITH_ENSENSO": self.options.with_ensenso,
+            "WITH_DAVIDSDK": self.options.with_davidsdk,
+            "WITH_DSSDK": self.options.with_dssdk,
+            "WITH_RSSDK": self.options.with_rssdk,
+            "PCL_SHARED_LIBS": self.options.shared,
+            "FLANN_USE_STATIC": not self.options["flann"].shared
+        }
+        pcl_features = {
+            "BUILD_kdtree": self.options.module_kdtree,
+            "BUILD_octree": self.options.module_octree,
+            "BUILD_search": self.options.module_search,
+            "BUILD_sample_consensus": self.options.module_sample_consensus,
+            "BUILD_filters": self.options.module_filters,
+            "BUILD_2d": self.options.module_2d,
+            "BUILD_geometry": self.options.module_geometry,
+            "BUILD_io": self.options.module_io,
+            "BUILD_features": self.options.module_features,
+            "BUILD_ml": self.options.module_ml,
+            "BUILD_segmentation": self.options.module_segmentation,
+            "BUILD_surface": self.options.module_surface,
+            "BUILD_module_registration": self.options.module_registration,
+            "BUILD_module_keypoints": self.options.module_keypoints,
+            "BUILD_module_tracking": self.options.module_tracking,
+            "BUILD_recognition": self.options.module_recognition,
+            "BUILD_stereo": self.options.module_stereo
+        }
+
+        if self.options.with_png:
+            pcl_config["LIBPNG_ROOT"] = self.deps_cpp_info["libpng"].rootpath
+        if self.options.with_qhull:
+            pcl_config["QHULL_ROOT"] = self.deps_cpp_info["qhull"].rootpath
+            pcl_config["QHULL_USE_STATIC"] = True
+
+        cmake = CMake(self)
+        cmake.definitions.update(cmake_definitions)
+        cmake.definitions.update(pcl_config)
+        cmake.definitions.update(pcl_features)
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        shutil.rmtree(os.path.join(self.package_folder, "share"))
+        shutil.rmtree(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "share"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

rmtree raises an exception if it's not possible to remove. Use conans.tools.rmdir instead.